### PR TITLE
Improve settings window

### PR DESCRIPTION
**Pull request by ericcornelissen**
I tried to improve the appearance and usability of the settings window mainly by using consistent margins, and add some quality of life improvements.

- [x] Normalized margins between settings
- [x] Always have some margin at the bottom of a page
- [x] Use constant width for text inputs
- [x] Added default display values for a Profile's "Working directory", "Arguments", and "Keyboard shortcut"
- [x] Attempt to make `.xaml` files more readable
- [x] When cancelling creating a new (or cloning) a profile, delete the new profile
- [x] Open "Create new theme" in edit mode
- [x] Don't prompt user for cancelling without changing anything in the Profile
- [x] Don't prompt user for cancelling without changing anything in the Themes
- [x] Update "loading..." text in About page

### CODE REVIEW

File: FluentTerminal.App/Views/SettingsPages/About.xaml [hunk 1]
```diff
@@ -52,36 +50,40 @@
...
-                    <TextBlock Visibility="{x:Bind ViewModel.LatestVersionFound, Mode=OneWay, Converter={StaticResource TrueToVisibleConverter}}">
-                        <Run Text="{x:Bind ViewModel.LatestVersion, Mode=OneWay}" />
-                         - <Hyperlink NavigateUri="{x:Bind ViewModel.LatestVersionReleaseNotesURL, Mode=OneWay}">
-                            Release notes
-                        </Hyperlink>
-                    </TextBlock>
+                    <StackPanel Margin="0,24,0,0">
+                        <TextBlock Margin="0,0,0,8" Text="Latest version" />
+                        <TextBlock Visibility="{x:Bind ViewModel.LatestVersionLoading, Mode=OneWay, Converter={StaticResource TrueToVisibleConverter}}">
+                            <Italic>loading...</Italic>
```
Comment (skipmeister123) [inline FluentTerminal.App/Views/SettingsPages/About.xaml]: Should this be capitalized? I feel like maybe "Checking for update..." would work better? Or maybe a spinner? I dunno. This just doesn't seem quite right.

Comment (ericcornelissen) [reply FluentTerminal.App/Views/SettingsPages/About.xaml]: That's a good a suggestion, I'll definitely update the text and I'll see about a spinner

**Comment by ericcornelissen** [comment]
What I did so far was manually update the margins of all elements in the `.xaml` files. Alternatively we could move this to a [XAML style](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/xaml-styles) which would have two major benefits:

1. Easier to change the overall margins
1. Contributors are less likely to make a mistake regarding margins in the future

Thoughts @felixse?

**Comment by skipmeister123** [comment]
Having this as a single variable seems like the best idea.

**Comment by felixse** [comment]
Good changes here, thank you 👍 

> Having this as a single variable seems like the best idea.


Yes, that actually how it should be done, just hacked this together initially and never bothered cleaning it up.

I have introduced two margin definitions:
ContainerMargin is defining margin left, right and bottom and should be used for the element that contains that items that should have equal margin.
ItemMargin is defining margin at the bottom, just for all direct children of the container.

Besides that I had some issues with the new Equals method in terminal theme, we should consider that author could also be null. Was running out of time tonight, but I can have a look at this tomorrow

File: FluentTerminal.App/Views/SettingsPages/About.xaml [hunk 2]
```diff
@@ -45,43 +45,46 @@
...
-                <StackPanel Grid.Row="1" Margin="0,72,0,24" VerticalAlignment="Bottom">
+                <StackPanel
+                    Grid.Row="1"
+                    Margin="0,72,0,0"
+                    VerticalAlignment="Bottom">
                     <TextBlock>
-                        Created by <Italic>felixse</Italic>
-                        and <Hyperlink NavigateUri="https://github.com/felixse/FluentTerminal/graphs/contributors">contributors</Hyperlink>
+                        Created by<Italic>felixse</Italic>
+                        and<Hyperlink NavigateUri="https://github.com/felixse/FluentTerminal/graphs/contributors">contributors</Hyperlink>
```
Comment (ericcornelissen) [inline FluentTerminal.App/Views/SettingsPages/About.xaml]: Don't think its intentional, but it seems you removed some necessary whitespace in this file by accident/automatically

File: FluentTerminal.App/Views/SettingsPages/About.xaml [hunk 3]
```diff
@@ -12,7 +12,7 @@
             <RowDefinition Height="Auto" />
             <RowDefinition Height="*" />
         </Grid.RowDefinitions>
-        <Grid Grid.Row="1" Margin="24,0,0,24">
+        <Grid Grid.Row="1" Margin="24,0,0,0">
```
Comment (ericcornelissen) [inline FluentTerminal.App/Views/SettingsPages/About.xaml]: Not as important, but we might consider defining a `<Thickness>` for this margin as well

**Comment by ericcornelissen** [comment]
> I have introduced two margin definitions [...]

I added a third one for the header margins just for completeness.


> Besides that I had some issues with the new Equals method in terminal theme, we should consider that author could also be null. Was running out of time tonight, but I can have a look at this tomorrow

I fixed this particular instance where a variable can be `null`. Alternatively, I think, we can instead use `Object.equals(obj1, obj2)` (see [docs](https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=netframework-4.7.2#System_Object_Equals_System_Object_System_Object_)) everywhere, but I'm not 100% that will work as intended, so I didn't update it yet...

File: FluentTerminal.App/Views/SettingsPages/About.xaml (hunk 2 above)
Comment (felixse) [reply FluentTerminal.App/Views/SettingsPages/About.xaml]: Damnit, not again. Xaml Styler is quite aggressive on whitespace. I will move the text to the code behind

**Pull request merged**